Spigot-API-Patches/0283-Add-TameableDeathMessageEvent.patch

```diff
@@ -0,0 +1,81 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Fri, 19 Mar 2021 23:25:38 -0400
+Subject: [PATCH] Add TameableDeathMessageEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/entity/TameableDeathMessageEvent.java b/src/main/java/io/papermc/paper/event/entity/TameableDeathMessageEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..518b2777a80f3964ec61b5828fe63d6dfed19765
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/entity/TameableDeathMessageEvent.java
+@@ -0,0 +1,69 @@
++package io.papermc.paper.event.entity;
++
++import net.kyori.adventure.text.Component;
++import org.bukkit.entity.Tameable;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Called when a {@link Tameable} dies and sends a death message.
++ */
++public class TameableDeathMessageEvent extends EntityEvent implements Cancellable {
++
++    private static final HandlerList handlers = new HandlerList();
++    private boolean cancelled;
++    private Component deathMessage;
++
++    public TameableDeathMessageEvent(@NotNull Tameable what, @NotNull Component deathMessage) {
++        super(what);
++        this.deathMessage = deathMessage;
++    }
++
++    /**
++     * Set the death message that appears to the owner of the tameable.
++     *
++     * @param deathMessage Death message to appear
++     */
++    public void setDeathMessage(@NotNull Component deathMessage) {
```

Comment:
Adventure API generally doesn't use get/set prefixes for methods

Reply:
I'm just curious is there a reason for that? (I noticed that too, and I will gladly change it but am just curious)

Reply:
not really sure! shorter lines of code, maybe? avoiding conflicts with existing code like PlayerDeathEvent#getDeathMessage? 🤷🏻‍♀️

Reply:
Yes, please make this change.

Reply:
Am I the only person that prefers the get/set prefixes so its easier to get a dropdown in an IDE of all available getters/setters just by typing the prefix? o_O

Reply:
I think I'm with billy on this. I do find myself using the get/set prefixes to see all the gettable/settable properties of an object.

Reply:
1. Leaving it as-is would make it inconsistent with existing code - all adventure methods lack those prefixes.
2. There's a few coexisting reasons for dropping those prefixes: Java getting rid of them (see Records and other new Java features), avoiding signature conflicts and Adventure lib API not having those methods (though there additional reason might be the "builders" being immutable, so methods are not traditional setters).

I feel like once the API decisions are solidified and it's merged, there's no reason to keep repeating your opinion as noone will rewrite whole adventure api integration just to bring older syntax back, so they bring no value.

Reply:
Indeed - please make the change as mentioned in the first comment, @Owen1212055.

Reply:
> Indeed - please make the change as mentioned in the first comment, @Owen1212055.

Yes I will, I'm sorry this comment was made a late and I'll be busy for a bit.